gtsam/discrete/discrete.i

```diff
@@ -62,20 +77,45 @@ virtual class DiscreteConditional : gtsam::DecisionTreeFactor {
       string s = "Discrete Conditional: ",
       const gtsam::KeyFormatter& formatter = gtsam::DefaultKeyFormatter) const;
   gtsam::DecisionTreeFactor* toFactor() const;
-  gtsam::DecisionTreeFactor* chooseAsFactor(const gtsam::DiscreteValues& parentsValues) const;
+  gtsam::DecisionTreeFactor* choose(
+      const gtsam::DiscreteValues& parentsValues) const;
+  gtsam::DecisionTreeFactor* likelihood(
+      const gtsam::DiscreteValues& frontalValues) const;
+  gtsam::DecisionTreeFactor* likelihood(size_t value) const;
   size_t solve(const gtsam::DiscreteValues& parentsValues) const;
   size_t sample(const gtsam::DiscreteValues& parentsValues) const;
-  void solveInPlace(gtsam::DiscreteValues@ parentsValues) const;
-  void sampleInPlace(gtsam::DiscreteValues@ parentsValues) const;
+  void solveInPlace(gtsam::DiscreteValues @parentsValues) const;
+  void sampleInPlace(gtsam::DiscreteValues @parentsValues) const;
   string markdown(const gtsam::KeyFormatter& keyFormatter =
+                      gtsam::DefaultKeyFormatter) const;
+};
+
+#include <gtsam/discrete/DiscretePrior.h>
+virtual class DiscretePrior : gtsam::DiscreteConditional {
+  DiscretePrior();
+  DiscretePrior(const gtsam::DecisionTreeFactor& f);
+  DiscretePrior(const gtsam::DiscreteKey& key, string spec);
+  void print(string s = "Discrete Prior\n",
+             const gtsam::KeyFormatter& keyFormatter =
                  gtsam::DefaultKeyFormatter) const;
+  double operator()(size_t value) const;
+  std::vector<double> pmf() const;
+  size_t solve() const;
+  size_t sample() const;
 };
 
 #include <gtsam/discrete/DiscreteBayesNet.h>
-class DiscreteBayesNet { 
+class DiscreteBayesNet {
   DiscreteBayesNet();
-  void add(const gtsam::DiscreteKey& key,
-           const gtsam::DiscreteKeys& parents, string spec);
+  void add(const gtsam::DiscreteConditional& s);
+  void add(const gtsam::DiscreteKey& key, string spec);
+  void add(const gtsam::DiscreteKey& key, string spec,
+           const gtsam::DiscreteKey& parent1);
+  void add(const gtsam::DiscreteKey& key, string spec,
+           const gtsam::DiscreteKey& parent1,
+           const gtsam::DiscreteKey& parent2);
+  void add(const gtsam::DiscreteKey& key, const gtsam::DiscreteKeys& parents,
```

Comment:
Why is this different?

Reply:
* the signature

Reply:
backwards compatible. I am now rethinking this: I'll make the specialization also specify the spec as the last argument.